### PR TITLE
[14.0][FIX] Add check_company=True on contracts and contract lines

### PR DESCRIPTION
**Pull request by alexis-via**
Add check_company=True in field definition and
[('company_id', '=', company_id)] in the domain (in view or in field definition)

### CODE REVIEW

**Review by mazairenouraly — APPROVED** [review]

**Review by leemannd — APPROVED** [review]

**Review by rousseldenis — APPROVED** [review]

File: contract_payment_mode/models/contract.py [hunk 1]
```diff
@@ -7,8 +7,9 @@ class ContractContract(models.Model):
     payment_mode_id = fields.Many2one(
         comodel_name="account.payment.mode",
         string="Payment Mode",
-        domain=[("payment_type", "=", "inbound")],
+        domain="[('payment_type', '=', 'inbound'), ('company_id', '=', company_id)]",
```
Comment (pedrobaeza) [inline contract_payment_mode/models/contract.py:10]: This should have pre-commit passed

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Comment by alexis-via** [comment]
Should be green and ready to merge

**Review by carlosdauden — APPROVED** [review]

File: contract_payment_mode/models/contract.py (hunk 1 above)
Comment (pedrobaeza) [reply contract_payment_mode/models/contract.py:10]: Please pass pre-commit, or this can't be merged.

**Comment by leemannd** [comment]
The pre-commit has been fixed and @pedrobaeza 's comment has been answered.

**Comment by pedrobaeza** [comment]
Please fw-port it to upper versions.

File: contract/models/abstract_contract.py [hunk 2]
```diff
@@ -22,7 +22,12 @@ class ContractAbstractContract(models.AbstractModel):
     partner_id = fields.Many2one(
         comodel_name="res.partner", string="Partner", index=True
     )
-    pricelist_id = fields.Many2one(comodel_name="product.pricelist", string="Pricelist")
+    pricelist_id = fields.Many2one(
+        comodel_name="product.pricelist",
+        string="Pricelist",
+        check_company=True,
```
Comment (MrTango) [inline contract/models/abstract_contract.py:28]: this leads to an issue with the pricelists in contract, when no multi companay is used.
I can no longer choose existing lists.
Before this change it was working fine.

File: contract/models/contract.py [hunk 3]
```diff
@@ -81,13 +84,19 @@ class ContractContract(models.Model):
     )
     date_end = fields.Date(compute="_compute_date_end", store=True, readonly=False)
     payment_term_id = fields.Many2one(
-        comodel_name="account.payment.term", string="Payment Terms", index=True
+        comodel_name="account.payment.term",
+        string="Payment Terms",
+        index=True,
+        check_company=True,
+        domain="[('company_id', '=', company_id)]",
```
Comment (marcelsavegnago) [inline contract/models/contract.py:91]: @alexis-via do you believe it is necessary to define this domain? In this case, the objective is to force a company to be defined in each payment_term? In theory, check_company would already check whether the company is false or equal to the company in the record.

**Pull request merged**